Marlin/src/inc/Conditionals_post.h

```diff
@@ -2166,7 +2166,7 @@
 #if PIN_EXISTS(Z4_MAX)
   #define HAS_Z4_MAX 1
 #endif
-#if BOTH(HAS_BED_PROBE, HAS_CUSTOM_PROBE_PIN) && PIN_EXISTS(Z_MIN_PROBE)
+#if HAS_BED_PROBE && PIN_EXISTS(Z_MIN_PROBE)
   #define HAS_Z_MIN_PROBE_PIN 1
```

Comment:
Many pins files just always define a `Z_MIN_PROBE_PIN`. We don't really want Marlin to see or use the pin at all unless the configuration specifies that it will be used for the bed probe, because it may prevent the pin being used for some other purpose. The `HAS_Z_MIN_PROBE_PIN` conditional is intended to reflect this. I assume that the `G38` probe is also the "`BED_PROBE`" since `HAS_BED_PROBE` is one of the requirements here.